mojave/structures/ruins.dm

```diff
@@ -0,0 +1,89 @@
+#define RUIN_BODIES_LEFT list("coupe-left" = 10, "muscle-left" = 3, "sport-left" = 3, "van-left" = 1)
+#define RUIN_BODIES_RIGHT list("coupe-right" = 10, "muscle-right" = 3, "sport-right" = 3, "van-right" = 1)
+
+/////////////////////////////////////////////////////////////
+////////////// MOJAVE SUN RUINS/TRASH ///////////////////////
+/////////////////////////////////////////////////////////////
+
+///Place here semi-functional structures that are basically for scrapping/joonking///
+
+//Cars//
+
+/obj/structure/ms13/vehicle
+	name = "vehicle ruin"
+	desc = "A base class unrandomised vehicle, if you see it, report it foo."
+	icon = 'mojave/icons/structure/vehicles.dmi'
+	icon_state = "helper"
+	density = TRUE
+	layer = FLY_LAYER
+	anchored = TRUE
+	bound_width = 96
+	var/direction = "test"
+	var/direction_base = "test"
+
+/obj/structure/ms13/vehicle/left
+	icon_state = "helper-left"
+	direction = "left"
+	direction_base = RUIN_BODIES_LEFT
+
+/obj/structure/ms13/vehicle/right
+	icon_state = "helper-right"
+	direction = "right"
+	direction_base = RUIN_BODIES_RIGHT
+
+/obj/structure/ms13/vehicle/Initialize()
+	. = ..()
+	var/randomiser = rand(0,8)
+	icon_state = pickweight(direction_base)
+	if(icon_state == "coupe-[direction]")
```

Comment:
Please do not use direction to encode variations in the sprite.  I see that direction in this case actually corresponds to color (for the body and various panels of the vehicle).  Please either change the color to explicit icon states or break up the dmi on a per color basis.  Left and right faces could also be properly encoded in direction instead of an explicit icon state.  It can't be merged in its current state.

Reply:
alright, recoded and fixed